packages/developer-console-spa/src/common/components/NewProjectForm/index.tsx

```diff
@@ -1,79 +1,85 @@
 import { ActionGroup, Button, Form, FormGroup, Modal, TextArea, TextInput } from '@patternfly/react-core';
 import { useEffect, useState } from 'react';
-import { useHistory, useLocation } from 'react-router-dom';
-import gqlClient from '../utils/gqlClient';
-import { newApp } from '../utils/gql-queries';
+import { useHistory } from 'react-router-dom';
+import gqlClient from 'common/utils/gqlClient';
+import { newProject } from 'common/utils/gql-queries';
 import { Controller, useForm } from 'react-hook-form';
 import { yupResolver } from '@hookform/resolvers/yup';
 import * as yup from 'yup';
+import useQueryParams from 'common/hooks/useQueryParams';
+import { useCallback } from 'react';
 
-interface IAppInput {
+interface IProjectInput {
   name: string;
   description: string;
 }
-const appSchema = yup.object().shape( {
+const projectSchema = yup.object().shape( {
   name: yup.string().trim().required(),
   description: yup.string().trim(),
 } );
 
-export default function AddAppForm () {
+export default function NewProjectForm () {
   const [ isOpen, setIsOpen ] = useState<boolean>();
-  const location = useLocation();
   const history = useHistory();
-  const { control, handleSubmit, formState: { errors, isValid }, reset } = useForm<IAppInput>( {
+  const searchParams = useQueryParams();
+  const { control, handleSubmit, formState: { errors, isValid }, reset } = useForm<IProjectInput>( {
     mode: 'onBlur',
-    resolver: yupResolver( appSchema ),
+    resolver: yupResolver( projectSchema ),
     defaultValues: {
       name: '',
       description: '',
     },
   });
 
   useEffect( () => {
```

Comment:
You dont need a useEffect for this. 

You can use a derived state, that is just use `const isOpen = searchParams.get('new') === 'true' `

And use the hook provided by react router to get query params

Reply:
Changed the `isOpen` to normal variable.

Btw, the `useSearchParams` hook is in react-router v6. This is not upgraded yet. Will do that in the next pr update.